src/ProjectOrigin.Wallet.IntegrationTests/TestClassFixtures/GrpcTestFixture.cs

```diff
@@ -0,0 +1,130 @@
+        private Action<IWebHostBuilder>? _configureWebHost;
+        private GrpcChannel? _channel;
+        private Dictionary<string, string?>? _configurationDictionary;
+
+        public event LogMessage? LoggedMessage;
+        public GrpcChannel Channel => _channel ??= CreateChannel();
+
+        public GrpcTestFixture()
+        {
+            LoggerFactory = new LoggerFactory();
+            LoggerFactory.AddProvider(new ForwardingLoggerProvider((logLevel, category, eventId, message, exception) =>
+            {
+                LoggedMessage?.Invoke(logLevel, category, eventId, message, exception);
+            }));
+        }
+
+        public void ConfigureWebHost(Action<IWebHostBuilder> configure)
+        {
+            _configureWebHost = configure;
+        }
+
+        public void ConfigureHostConfiguration(Dictionary<string, string?> configuration)
+        {
+            _configurationDictionary = configuration;
+        }
+
+        private void EnsureServer()
+        {
+            if (_host == null)
+            {
+                var builder = new HostBuilder()
+                    .ConfigureServices(services =>
+                    {
+                        services.AddSingleton<ILoggerFactory>(LoggerFactory);
+                    })
+                    .ConfigureWebHostDefaults(webHost =>
+                    {
+                        webHost
+                            .UseTestServer()
+                            .UseEnvironment("Development")
+                            .UseStartup<TStartup>();
+
+                        _configureWebHost?.Invoke(webHost);
+                    });
+
+                if (_configurationDictionary != null)
+                {
+                    builder.ConfigureHostConfiguration(config =>
+                        {
+                            config.Add(new MemoryConfigurationSource()
+                            {
+                                InitialData = _configurationDictionary
+                            });
+                        });
+                }
+
+                _host = builder.Start();
+                _server = _host.GetTestServer();
+                _handler = _server.CreateHandler();
+            }
+        }
+
+        public LoggerFactory LoggerFactory { get; }
+
+        private GrpcChannel CreateChannel()
+        {
+            EnsureServer();
+
+            return GrpcChannel.ForAddress(_server!.BaseAddress, new GrpcChannelOptions
+            {
+                LoggerFactory = LoggerFactory,
+                HttpHandler = _handler
+            });
+        }
+
+        public void Dispose()
+        {
+            _channel?.Dispose();
+            _handler?.Dispose();
+            _host?.Dispose();
+            _server?.Dispose();
+        }
+
+        public IDisposable GetTestContext(ITestOutputHelper outputHelper)
+        {
+            return new GrpcTestContext<TStartup>(this, outputHelper);
+        }
```

Comment:
Is the GrpcTestContext needed? This method is never called and it is the only place where the GrpcTestContext is constructed.